### PR TITLE
refactor(HMS-3527): mock rbac standalone

### PR DESCRIPTION
**Pull request by avisiedo**
Refactor the rbac mock to get a standalone service and
update local infrastructure to run it as a container.

The container is available at: [quay.io/podengo/mock-rbac:1.0.0](https://quay.io/repository/podengo/mock-rbac?tab=tags).

Depends on: https://github.com/podengo-project/idmsvc-backend/pull/163

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -14,6 +14,14 @@
             "args": [],
             "env": {
                 "CONFIG_PATH": "${workspaceFolder}/configs",
+                "ENV_NAME": "local",
```
Comment (tiran) [inline .vscode/launch.json:17]: The variable name `ENV_NAME` is rather generic. What are other possible values? Is it just for RBAC testing or is it used for other settings, too?

File: build/mock-rbac/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+##
+FROM registry.access.redhat.com/ubi9/go-toolset:1.20 as builder
+LABEL idmsvc-backend=builder
+# https://developers.redhat.com/articles/2022/05/31/your-go-application-fips-compliant
+ENV OPENSSL_FORCE_FIPS_MODE=1
+WORKDIR /go/src/app
+COPY . .
+USER 0
+RUN git log -1
+RUN make get-deps bin/mock-rbac
```
Comment (tiran) [inline build/mock-rbac/Dockerfile:12]: Do we need `get-deps`? Does the container build work without it?
```suggestion
RUN make bin/mock-rbac
```

File: build/mock-rbac/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.3-1552
+LABEL idmsvc-backend=backend
+# https://developers.redhat.com/articles/2022/05/31/your-go-application-fips-compliant
+ENV OPENSSL_FORCE_FIPS_MODE=1
+RUN mkdir -p /opt/bin /opt/bin/scripts/db /opt/bin/configs
+WORKDIR /opt/bin
+COPY --from=builder /go/src/app/bin/mock-rbac ./
+COPY scripts/db/migrations /opt/bin/scripts/db/migrations
```
Comment (tiran) [inline build/mock-rbac/Dockerfile:22]: Why do we need DB migrations in the mock container?

File: internal/config/config.go [hunk 4]
```diff
@@ -328,6 +328,17 @@ func Load(cfg *Config) *viper.Viper {
...
+		reportError(err)
+		panic("Invalid configuration")
+	}
+
+	sec, err := secrets.NewAppSecrets(cfg.Application.MainSecret)
+	if err != nil {
+		panic(err)
+	}
+	cfg.Secrets = *sec
+
```
Comment (tiran) [inline internal/config/config.go:341]: As discussed in the other PR, this section should not be moved.

**Review by tiran — CHANGES_REQUESTED** [review]

File: build/mock-rbac/Dockerfile (hunk 3 above)
Comment (avisiedo) [reply build/mock-rbac/Dockerfile:22]: thanks! we don't need it! "copy&paste"

File: .vscode/launch.json (hunk 1 above)
Comment (avisiedo) [reply .vscode/launch.json]: I was wrong! I have updated the PR with this change! thanks!

File: internal/config/config.go (hunk 4 above)
Comment (avisiedo) [reply internal/config/config.go:341]: thanks! Restoring!

**Review by frasertweedale — APPROVED** [review]
Tested locally, tests are passing after the config change (which is mentioned in comments and in the commit message - thanks!).  So, let's merge it.

**Pull request merged**